### ISSUE TITLE
Proposals for enhancing fight mechanics

### ISSUE THREAD

**Issue by Gandifil** [devapromix/troll-temple#4]
I have some proposals for fight mechanics and damage calculating!

1.  Split damage on two types: physical and magic(elemental). We can draw this types by different colors. We can make special weapon with elemental damage.
2. Enhance defence mechanics. 3 type: evasion, armor and resist. Evasion will ability to evade attack and skip absolutely damage. Armor decrease damage. Resist can ignore some percent of damage.  Armor - for physical, resist - for elemental, evasion - for both.
3. Perks. Abilities, which are presents at levelup give some bonuses. When player get levelup, it can choose one from N random perks.
-  Perks can get and simple attributes bonus (+50 max hp), and new "deep" abilities/machanics (for example, use two weapon in one time)
- Perks can have negative bonuses for more cool building character (for example - "you can't do critical strikes, +40% damage)
- Perks have different rarity, for example, "usuall" and "rare". Every levelup player have a small chance to get rare perk, but every 5th levelup (5, 10, 15, etc) all perks are rare.

**Comment by devapromix** [comment]
Agree. It’s possible to do that too. Elemental magic can also be divided into ice, fire and electricity. And add poison mechanics.
